aya/src/programs/sock_ops.rs

```diff
@@ -58,9 +72,16 @@ impl SockOps {
     ///
     /// The returned value can be used to detach, see [SockOps::detach].
     pub fn attach<T: AsFd>(&mut self, cgroup: T) -> Result<SockOpsLinkId, ProgramError> {
+        self.attach_with_flags(cgroup, SockOpsFlags::empty())
+    }
+
+    /// Attaches the program to the given cgroup.
+    ///
+    /// The returned value can be used to detach, see [SockOps::detach].
+    pub fn attach_with_flags<T: AsFd>(&mut self, cgroup: T, flags: SockOpsFlags) -> Result<SockOpsLinkId, ProgramError> {
```

Comment:
I think I'd rather break the API here and add `flags: CgroupAttachFlags`.
This makes it explicit which flags are being used.

Reply:
Sounds good.  I wasn't sure how much of an expanding change you'd prefer.  I've modified the `attach()` fn signature here and for other program types.

Reply:
> This makes it explicit which flags are being used.

I'm in two minds about this. For the `Cgroup*` and `SockOps` programs, I think it makes sense to have to pass `CgroupAttachFlags` to `attach()` - you're explicitly working with cgroups, so it makes sense to _have_ to think about what's going on with other cgroups.

But for `SkMsg` `SkSkb` and  `LircMode2`, I think we should default to no flags like before, and have a separate `attach_with_flags()` for the cases in which you do care about cgroups.

Reply:
That's a keen observation.  I've updated this change following your advice to extend `attach()` for specific program types, and adding `attach_with_flags()` where the developer does not necessarily need to think about cgroups.